pom.xml

```diff
@@ -706,7 +802,12 @@
                         </configuration>
                     </execution>
                     <execution>
-                        <id>default-cli</id>
+                        <!-- Do a merge after unit tests complete, to cover the case that only goals up to 'test' phase
```

Comment:
This may be ok, but it may also cause duplicate processing of data in the test report.  The merging needs to take place after ALL tests complete.

Adding the command line configuration does something different.  It allows you to run the jacoco:merge goal from the command line and have it work as expected.  It needs to stay in place.

Reply:
just to clarify, as long as the tests are passing, the only change I have requested is that the default-cli execution for jacoco:merge be restored.

Reply:
> just to clarify, as long as the tests are passing, the only change I have requested is that the default-cli execution for jacoco:merge be restored.

I see; when run from commandline directly as `jacoco:merge` (as you suggested in #8457), the configuration is missing - an error happens. I beg your pardon, but given that Jacoco coverage runs automatically (during tests) - the user does not enter anything, just `mvn test`, and with my fix, `jacoco:merge` runs *also* automatically during the tests ....
... what is the use-case the user would **maually** call `mvn jacoco:merge` ? I just couldn't find a use-case.

To clarify the "removal" : The original `default-cli` execution ID binds to `generate-resources` if *no phase is present*; that means the `merge` goal is run before **any** of the tests can be run. 
An **additional** `merge` runs if one uses command line suggested in #8457. On unclean builds (without explicit clean), 
* the data from the *previous* test run are merged, 
* tests are run
* the data from *this* test run are merged.

**The execution from master branch**:
```mvn -B test integration-test failsafe:verify```  (taken from `travis.sh`) yields:
```
[INFO] Scanning for projects...
...
[INFO] --- jacoco-maven-plugin:0.8.5:prepare-agent (prepare-agent) @ jmri ---
[INFO] --- jacoco-maven-plugin:0.8.5:prepare-agent (prepare-agent-integration) @ jmri ---
[INFO] --- jacoco-maven-plugin:0.8.5:merge (default-cli) @ jmri ---
>>> [INFO] Loading execution data file /space/src/jmri/fixes2/JMRI/target/jacoco-unit.exec
>>> [INFO] Loading execution data file /space/src/jmri/fixes2/JMRI/target/jacoco-it.exec
>>> [INFO] Loading execution data file /space/src/jmri/fixes2/JMRI/target/jacoco.exec
>>> [INFO] Writing merged execution data to /space/src/jmri/fixes2/JMRI/target/jacoco.exec
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ jmri ---
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ jmri ---
...
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ jmri ---
...
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ jmri ---
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] --- build-helper-maven-plugin:3.0.0:timestamp-property (timestamp-property) @ jmri ---
[INFO] --- jacoco-maven-plugin:0.8.5:prepare-agent (prepare-agent) @ jmri ---
[INFO] --- jacoco-maven-plugin:0.8.5:prepare-agent (prepare-agent-integration) @ jmri ---
[INFO] --- jacoco-maven-plugin:0.8.5:merge (default-cli) @ jmri ---
>>> [INFO] Loading execution data file /space/src/jmri/fixes2/JMRI/target/jacoco-unit.exec
>>> [INFO] Loading execution data file /space/src/jmri/fixes2/JMRI/target/jacoco-it.exec
>>> [INFO] Loading execution data file /space/src/jmri/fixes2/JMRI/target/jacoco.exec
>>> [INFO] Writing merged execution data to /space/src/jmri/fixes2/JMRI/target/jacoco.exec
[INFO] --- maven-failsafe-plugin:2.22.0:integration-test (default) @ jmri ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
...
[INFO] --- maven-failsafe-plugin:2.22.0:verify (default-cli) @ jmri ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------

```
Note that the `.exec` files may be actually merged *twice*. Also note that `mvn integration-test` **does not** merge in `jacoco-it.exec` after integration tests complete, and before `failsafe:verify` target.

If you have concerns about merging the data sets twice, what about this twice merged, but not finalized (with integration test results) data, introduced by #8457 ?

It's probably because the `jacoco:report` binds by default to `verify` phase.

I am not a travis or jmri build expert, sorry. Can someone explain me where the `mvn verify` or specifically the `jacoco:check` target is executed during the JMRI travis builds ?

Reply:
> I am not a travis or jmri build expert, sorry. Can someone explain me where the `mvn verify` or specifically the `jacoco:check` target is executed during the JMRI travis builds ?

Sorry, found: `mvn verify` in one of `travis.sh` branches - good ! So there's a 3rd merge before `jacoco:report` in the master.

Apologies: the CI merges everything in time. It's just run `mvn test integration-test` which does not (in the master).